freedomofpress/securedrop.org
# 446 reset migration history

Reset migration history:

* Delete old migration files and generate fresh ones based on the state of the model files
* Introduce a `clearmigrationhistory` command that deletes rows from the `django_migrations` table
* Delete the vestigial `landing_page_checker` app

## How to test:

1. Ensure that your database is up to date with the latest migrations on `master`. A good way to do that:

   ```python
   git checkout master
   make dev-import-db # Optional: import a recent db dump, if you have one
   docker exec -it sd_django ./manage.py migrate
   ```

   If you like, at this point, verify that the `django_migrations` table includes lots of records for migrations on securedrop.org-specific apps.

2. From this branch, clear the migrations table:

   ```python
   git checkout 446-reset-migration-history
   docker exec -it sd_django ./manage.py clearmigrationhistory
   ```

   Confirm that you get _exactly_ this output:

   ```
   Deleted 70 migration records in 11 apps
   ```

   If you like, at this point, verify that the `django_migrations` table includes no records for migrations on securedrop.org-specific apps (it will still include migrations for built-in and third-party models).

3. From this branch, fake the new migration history:

   ```sh
   docker exec -it sd_django ./manage.py migrate --fake
   ```

   If you like, at this point, verify that the `django_migrations` table includes one or two migrations for each securedrop.org-specific app.

   Make sure the dev server is still up. Run `make dev-go` if it isn't. Click around the site and the admin and make sure everything is still behaving properly.

   For bonus extra thoroughness, experiment with adding a field to a model and running

   ```sh
   docker exec -it sd_django ./manage.py makemigrations
   ```

   and

   ```sh
   docker exec -it sd_django ./manage.py migrate
   ```

   to verify that everything is behaving properly.

## How to deploy:

After merge, follow the instructions in #446
